### PR TITLE
Java 8 migration. Use string switch where possible

### PR DESCRIPTION
**Pull request by doom369**
Motivation:

Replace "if else" conditions with string switch. It is easier to read the code, for large "if else" constructions switch also could be faster.

Modification:

Replaced "if else" with a string switch.

### CODE REVIEW

File: handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java [hunk 1]
```diff
@@ -1515,48 +1515,55 @@ public final void setEnabledProtocols(String[] protocols) {
...
+                    }
+                    break;
+                case PROTOCOL_TLS_V1_3:
+                    if (minProtocolIndex > OPENSSL_OP_NO_PROTOCOL_INDEX_TLSv1_3) {
+                        minProtocolIndex = OPENSSL_OP_NO_PROTOCOL_INDEX_TLSv1_3;
+                    }
+                    if (maxProtocolIndex < OPENSSL_OP_NO_PROTOCOL_INDEX_TLSv1_3) {
+                        maxProtocolIndex = OPENSSL_OP_NO_PROTOCOL_INDEX_TLSv1_3;
+                    }
+                    break;
```
Comment (normanmaurer) [inline handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java]: @doom369 can you also add a default ?

Comment (doom369) [reply handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java]: @normanmaurer I think this is should go in separate PR. This PR one doesn't change existing behavior.

Comment (normanmaurer) [reply handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java]: true...

**Comment by normanmaurer** [comment]
@netty-bot test this please

**Comment by normanmaurer** [comment]
@doom369 thanks!

**Pull request merged**